lib/models/auth_token.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'auth_token.g.dart';
+
+@JsonSerializable()
+class AuthToken {
+  @JsonKey(name: 'access_token')
+  String accessToken;
+  @JsonKey(name: 'token_type')
+  String tokenType;
```

Comment:
What do you think about creating an enum for this?
```dart
enum AuthTokenType {
  @JsonValue("bearer") BEARER
  ...
}
```

Reply:
eventually I will store both of this as a pair and just use it...the front-end doesn't need to know how many type the API supposes to return imo, so capturing them with enum is kinda useless here (single value and we don't have any idea about whatever the other token type)

Reply:
just for ur clarity, we may just store to shared_pref and retrieve as `{token_type} {token}` so I don't think we need the enum in play here 🤙

Reply:
I mean that if we setup a type here, it will more convenient to use and refactor. Such as comparing. also prevent the other hard compare it with raw string, IMO 😁 
```dart
info.tokenType == "ABC" // bad
info.tokenType == AUTH_TOKEN_TYPE_ABC // not cool enough
info.tokenType == AUTH_TOKEN_TYPE.ABC // 💯 
```

>the front-end doesn't need to know how many type the API supposes

Total agree, and we only need to add what we need to handle and fallback a default value for the others

Reply:
@sleepylee ahh got your point here. So a string here is totaly make sense 👍